echopype/mask/api.py

```diff
@@ -493,23 +495,72 @@ def frequency_differencing(
+        lhs = source_Sv["Sv"].sel(channel=chanA) - source_Sv["Sv"].sel(channel=chanB)
 
-    # assign a name to DataArray
-    da.name = "mask"
+        # create mask using operator lookup table
+        da = xr.where(str2ops[operator](lhs, diff), True, False)
 
-    # assign provenance attributes
-    mask_attrs = {"mask_type": "frequency differencing"}
-    history_attr = (
-        f"{datetime.datetime.utcnow()} +00:00. "
-        "Mask created by mask.frequency_differencing. "
-        f"Operation: Sv['{chanA}'] - Sv['{chanB}'] {operator} {diff}"
-    )
+        # assign a name to DataArray
+        da.name = "mask"
+
+        # assign provenance attributes
+        mask_attrs = {"mask_type": "frequency differencing"}
+        history_attr = (
+            f"{datetime.datetime.utcnow()} +00:00. "
+            "Mask created by mask.frequency_differencing. "
+            f"Operation: Sv['{chanA}'] - Sv['{chanB}'] {operator} {diff}"
+        )
 
-    da = da.assign_attrs({**mask_attrs, **{"history": history_attr}})
+        da = da.assign_attrs({**mask_attrs, **{"history": history_attr}})
 
-    return da
+        return da
+    # If Sv data is dask array
+    else:
+        num_column_blocks = int(np.ceil(source_Sv["Sv"].shape[2] / source_Sv["Sv"].chunks[2][0]))
+
+        @dask.delayed
+        def _frequency_differencing(sv_block, freqA_pos: int, freqB_pos: int):
+            # sv_block is ndarray.
+            return sv_block[freqA_pos] - sv_block[freqB_pos]
+
+        @dask.delayed
+        def _masking(lhs, diff: float):
+            return dask.array.where(str2ops[operator](lhs, diff), True, False)
+
+        da_list = []
+        # Iterate over all the chunks
+        for c in source_Sv["Sv"].data.blocks.ravel():
+            lhs = _frequency_differencing(c, freqA_pos, freqB_pos)
+            da_delayed = _masking(lhs, diff)
+            da_list.append(
+                # Convert dask delayed fn to dask array
+                dask.array.from_delayed(da_delayed, shape=(c.shape[1], c.shape[2]), dtype=c.dtype)
+            )
+
+        # Concatenate the dask chunks.
+        da_output = []
+        for i in range(0, len(da_list), num_column_blocks):
+            da_output.append(dask.array.concatenate(da_list[i : i + num_column_blocks], axis=1))
+
+        da_output_concat = dask.array.concatenate(da_output, axis=0)
+
+        return xr.DataArray(
+            name="mask",
+            data=da_output_concat,
+            coords={
+                "ping_time": source_Sv["Sv"].coords["ping_time"].values,
+                "range_sample": source_Sv["Sv"].coords["range_sample"].values,
+            },
+            attrs={
+                "mask_type": "frequency differencing",
+                "history": f"{datetime.datetime.utcnow()} +00:00. "
+                "Mask created by mask.frequency_differencing. "
+                f"Operation: Sv['{chanA}'] - Sv['{chanB}'] {operator} {diff}",
+            },
```

Comment:
This is similar to assign attrs above

https://github.com/OSOceanAcoustics/echopype/pull/1198/files#diff-36c0af5a3cf3a0fa59c1880a35e59e23e136740d1b963ffda24d8c714989879dR512-R520

I suggest extracting this out from the `if-else`

Reply:
I extracted that out. 

I think `datetime.datetime.utcnow()` might be different than what was being computed earlier because that value will be calculated before the masking operations since it's outside of `if-else.`